external/fv3fit/fv3fit/reservoir/domain2.py

```diff
@@ -306,7 +306,7 @@ def merge_all_subdomains(self, data: np.ndarray) -> np.ndarray:
             dim_slices = self._add_potential_leading_dim_to_slices(
                 subdomain.shape, dim_slices
             )
-            merged[dim_slices] = subdomain
+            merged[tuple(dim_slices)] = subdomain
```

Comment:
This addresses a FutureWarning that using non-tuple indices may lead to unexpected results

Reply:
Thank you!